### PR TITLE
Streaming improve schema inference support.

### PR DESCRIPTION
**Pull request by rozza**
Support inferred schemas when
`change.stream.publish.full.document.only = true`

SPARK-382

### CODE REVIEW

File: src/integrationTest/java/com/mongodb/spark/sql/connector/read/MongoBatchTest.java [hunk 1]
```diff
@@ -285,9 +290,12 @@ void testReadsAreSupportedWithSchemaInferred() {
...
             asList(
-                createStructField("_id", DataTypes.IntegerType, true),
-                createStructField("misc", DataTypes.StringType, true),
-                createStructField("name", DataTypes.StringType, true)));
+                createStructField(
+                    "_id", DataTypes.IntegerType, true, InferSchema.INFERRED_METADATA),
+                createStructField(
+                    "misc", DataTypes.StringType, true, InferSchema.INFERRED_METADATA),
+                createStructField(
+                    "name", DataTypes.StringType, true, InferSchema.INFERRED_METADATA)));
```
Comment (katcharov) [inline src/integrationTest/java/com/mongodb/spark/sql/connector/read/MongoBatchTest.java:298]: Should there be an override that takes true and INFERRED_METADATA?

**Review by katcharov — APPROVED** [review]
LGTM, just a couple of nits

**Review by vbabanin — APPROVED** [review]

**Comment by vbabanin** [comment]
LGTM!

**Comment by rozza** [comment]
Manually rebased and merged.

**Pull request closed without merging**